packages/noya-designsystem/src/components/InputField.tsx

```diff
@@ -1,59 +1,108 @@
+import { DropdownMenu as NoyaDropdownMenu } from 'noya-designsystem';
+import Button from './Button';
 
 type LabelPosition = 'start' | 'end';
 
 const InputFieldContext = createContext<{
   labelPosition: LabelPosition;
   labelSize: number;
+  hasDropdown: boolean;
 }>({
   labelPosition: 'end',
   labelSize: 6,
+  hasDropdown: false,
 });
 
 /* ----------------------------------------------------------------------------
  * Label
  * ------------------------------------------------------------------------- */
 
-const LabelContainer = styled.label<{ labelPosition: LabelPosition }>(
-  ({ theme, labelPosition }) => ({
-    color: theme.colors.textMuted,
-    position: 'absolute',
-    top: 0,
-    right: 0,
-    bottom: 0,
-    left: 0,
-    display: 'flex',
-    alignItems: 'center',
-    pointerEvents: 'none',
-    fontWeight: 'bold',
-    fontSize: '60%',
-    opacity: 0.5,
-    userSelect: 'none',
-    ...(labelPosition === 'start'
-      ? { justifyContent: 'flex-start', paddingLeft: '6px' }
-      : { justifyContent: 'flex-end', paddingRight: '6px' }),
-  }),
-);
+const LabelContainer = styled.label<{
+  labelPosition: LabelPosition;
+  hasDropdown: boolean;
+}>(({ theme, labelPosition, hasDropdown }) => ({
+  color: theme.colors.textMuted,
+  position: 'absolute',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  display: 'flex',
+  alignItems: 'center',
+  pointerEvents: 'none',
+  fontWeight: 'bold',
+  fontSize: '60%',
+  opacity: 0.5,
+  userSelect: 'none',
+  ...(labelPosition === 'start'
+    ? { justifyContent: 'flex-start', paddingLeft: '6px' }
+    : {
+        justifyContent: 'flex-end',
+        paddingRight: hasDropdown ? '16px' : '6px',
+      }),
+}));
 
 interface InputFieldLabelProps {
   children?: ReactNode;
 }
 
 function InputFieldLabel({ children = false }: InputFieldLabelProps) {
-  const { labelPosition } = useContext(InputFieldContext);
+  const { labelPosition, hasDropdown } = useContext(InputFieldContext);
+
+  return (
+    <LabelContainer labelPosition={labelPosition} hasDropdown={hasDropdown}>
+      {children}
+    </LabelContainer>
+  );
+}
+
+/* ----------------------------------------------------------------------------
+ * Dropdown
+ * ------------------------------------------------------------------------- */
+
+const DropdownContainer = styled.span(({ theme }) => ({
+  position: 'absolute',
+  right: 0,
+}));
+
+interface InputFieldDropdownProps {
+  list: { value: number; title: string }[];
+  onSumbitList: (value: string) => void;
+}
 
+function InputFieldDropdownMenu({
+  list,
+  onSumbitList,
+}: InputFieldDropdownProps) {
   return (
-    <LabelContainer labelPosition={labelPosition}>{children}</LabelContainer>
+    <DropdownContainer>
+      <NoyaDropdownMenu.Root<string>
+        items={list.map((e) => ({
+          value: e.value.toString(),
+          title: e.title,
+        }))}
+        onSelect={onSumbitList}
+      >
+        <Button id={'a'} variant="thin">
```

Comment:
`id` will need to be a prop, since this element is in a list, and every HTML id must be globally unique